### DISCUSSION TITLE
How to define variables myself?

### DISCUSSION THREAD

**Discussion by ghost** [creativeprojects/resticprofile#302] [Q&A] [answered]
I followed the instruction on [Variables :: resticprofile](https://creativeprojects.github.io/resticprofile/configuration/variables/index.html#hand-made-variables), but it did not work.

`profiles.toml`:

```toml
{{ $tmp := 'D:\tmp'}}

[default.env]
TMP = '{{ $tmp }}'
RESTIC_PASSWORD = '123'
RESTIC_REPOSITORY = '.\repo'
```

```
resticprofile --dry-run init
2024/01/29 10:38:33 using configuration file: profiles.toml
2024/01/29 10:38:33 cannot load configuration file: cannot compile template: __config:profiles.toml:1: malformed character constant: 'D:\tmp'
```

```
resticprofile version
resticprofile version 0.24.0 commit bcfaaa7cacfa7f88a94d5dc62ca69b2d89205d89
```

**Comment by creativeprojects** [comment]
Sorry for the late reply:

A string in go templates can only be declared using double quotes, it doesn't work with single quotes.

Try this instead:

```
{{ $tmp := "D:\tmp"}}
```

**Reply by ghost** [reply]
`{{ $tmp := "D:\tmp"}}` works, but `{{ $tmp := "D:\a\tmp"}}` doesn't work:

```
resticprofile show
2024/02/06 06:39:25 using configuration file: profiles.toml
2024/02/06 06:39:25 cannot load configuration file: cannot parse toml configuration: While parsing config: toml: invalid character
```

**Reply by creativeprojects** [reply]
I don't have access to a Windows machine right now, but in the meantime you can try these methods:

```
{{ $tmp := "D:\\a\\tmp"}}  (negates the effect of the \ to form special characters)
{{ $tmp := "D:/a/tmp"}}    (forward slash is allowed in go code)
{{ $tmp := `D:\a\tmp`}}    (use the backtick)
```

I'll update the documentation after finding out the best way 😉

**Reply by ghost** [reply]
They all work!

**Reply by creativeprojects** [reply, marked as answer]
I have updated the documentation accordingly:

https://7a57af69.resticprofile.pages.dev/configuration/variables/#hand-made-variables
